### PR TITLE
Matrix groups: cleanup some use of frobenius

### PR DESCRIPTION
**Pull request by fingolfin**
Also added a comment on how conjugate_transpose could be
made more efficient if needed

### CODE REVIEW

**Review by mjrodgers — APPROVED** [review]
I think this looks good, I can talk to @fieker about updating to precompute the Frobenius map if this is an improvement we should make but in the meantime I think this should be merged (modulo the tests passing, looking at those now).

File: src/Groups/matrices/transform_form.jl [hunk 1]
```diff
@@ -75,17 +79,17 @@ function _block_anisotropic_elim(B::MatElem{T}, _type::Symbol; isotr=false, f=0)
...
+      star(X) = transpose(X)
    elseif _type==:alternating
       degF=0
       s=-1
+      star(X) = transpose(X)
    elseif _type==:hermitian
       degF=div(degree(F),2)
       s=1
+      star(X) = conjugate_transpose(X)
    end
```
Comment (joschmitt) [inline src/Groups/matrices/transform_form.jl]: This doesn't seem to be legal julia code :( From one of the CI runners:
```
┌ Oscar
│  WARNING: Method definition star(Any) in module Oscar at /home/runner/work/Oscar.jl/Oscar.jl/src/Groups/matrices/transform_form.jl:82 overwritten at /home/runner/work/Oscar.jl/Oscar.jl/src/Groups/matrices/transform_form.jl:86.
│  ERROR: Method overwriting is not permitted during Module precompilation. Use `__precompile__(false)` to opt-out of precompilation.
└ 
```

Comment (thofma) [reply src/Groups/matrices/transform_form.jl]: You can replace those by `star = X -> conjugate_transpose(X)` etc and it should work.

Comment (mjrodgers) [reply src/Groups/matrices/transform_form.jl]: Yeah, but we still have an error in the test file; for some reason even though we have a symmetric form over GF(3), it is trying to call `conjugate_transpose` and I'm trying to figure out why. It comes from here:
```julia
R = residue_ring(ZZ, ZZ(3))[1]
F = GF(3)
G = diagonal_matrix([R(1), R(1)])
L = Oscar._orthogonal_grp_gens_odd(G, 3)
```

Comment (joschmitt) [reply src/Groups/matrices/transform_form.jl]: I pushed a commit, let's see.

Comment (joschmitt) [reply src/Groups/matrices/transform_form.jl]: @mjrodgers I didn't see your answer in time. Maybe the error is related to the failed precompilation?

Comment (mjrodgers) [reply src/Groups/matrices/transform_form.jl]: no, your push fixed the precompilation but this error is still there. it gives
```julia
ERROR: MethodError: no method matching _block_anisotropic_elim(::FqMatrix, ::Symbol)

Closest candidates are:
  _block_anisotropic_elim(::MatElem{T}, ::Val{_type}; isotr, f) where {T<:FinFieldElem, _type}
   @ Oscar ~/.julia/dev/Oscar/src/Groups/matrices/transform_form.jl:71

Stacktrace:
  [1] _block_anisotropic_elim(B::FqMatrix, ::Val{:symmetric}; isotr::Bool, f::Int64)
    @ Oscar ~/.julia/dev/Oscar/src/Groups/matrices/transform_form.jl:150
  [2] _block_anisotropic_elim(B::FqMatrix, ::Val{:symmetric})
    @ Oscar ~/.julia/dev/Oscar/src/Groups/matrices/transform_form.jl:71
  [3] _block_herm_elim(B::FqMatrix, _type::Symbol)
    @ Oscar ~/.julia/dev/Oscar/src/Groups/matrices/transform_form.jl:190
  [4] _change_basis_forms(B1::FqMatrix, B2::FqMatrix, _type::Symbol)
    @ Oscar ~/.julia/dev/Oscar/src/Groups/matrices/transform_form.jl:333
  [5] is_congruent(f::SesquilinearForm{FqFieldElem}, g::SesquilinearForm{FqFieldElem})
    @ Oscar ~/.julia/dev/Oscar/src/Groups/matrices/transform_form.jl:425
  [6] is_congruent(f::SesquilinearForm{FqFieldElem}, g::SesquilinearForm{FqFieldElem})
    @ Oscar ~/.julia/dev/Oscar/src/Groups/matrices/transform_form.jl:409
  [7] isometry_group(f::SesquilinearForm{FqFieldElem})
    @ Oscar ~/.julia/dev/Oscar/src/Groups/matrices/form_group.jl:621
  [8] _gens_form(G::ZZModMatrix, form_constructor::typeof(quadratic_form), p::Int64)
    @ Oscar ~/.julia/dev/Oscar/src/Groups/matrices/FiniteFormOrthogonalGroup.jl:124
  [9] _orthogonal_grp_gens_odd(G::ZZModMatrix, p::Int64)
    @ Oscar ~/.julia/dev/Oscar/src/Groups/matrices/FiniteFormOrthogonalGroup.jl:149
 [10] top-level scope
    @ REPL[5]:1
```

Comment (mjrodgers) [reply src/Groups/matrices/transform_form.jl]: it seems to give a different error now, but is not fixed

Comment (mjrodgers) [reply src/Groups/matrices/transform_form.jl]: Okay, I think it is fixed with that last commit I pushed.

**Comment by mjrodgers** [comment]
Have another error I'm trying to straighten out still, getting a `BoundsError` from `conjugate_transpose` with this one:
```julia
F,a = finite_field(3,2,"a")
x = diagonal_matrix(F.([1,2,1,1,1]))
y = diagonal_matrix(F.([2,1,0,0,2]))
y[3,4]=a; y[4,3]=a^3; y[4,5]=1+a; y[5,4]=(1+a)^3;
f = hermitian_form(x); g = hermitian_form(y)
is_true,z = is_congruent(f,g)
```

**Comment by mjrodgers** [comment]
> Have another error I'm trying to straighten out

OK I think this is resolved too, if the rest of the tests pass I think this can be merged now.

**Comment by fingolfin** [comment]
Thank you all for brining my shoddy PR over the finishing line :-)

**Pull request merged**